internal/compiler/expression_tree.rs

```diff
@@ -577,13 +577,17 @@ impl Expression {
             Expression::ElementReference(_) => Type::ElementReference,
             Expression::RepeaterIndexReference { .. } => Type::Int32,
             Expression::RepeaterModelReference { element } => {
-                if let Expression::Cast { from, .. } = element
-                    .upgrade()
-                    .unwrap()
-                    .borrow()
-                    .repeated
-                    .as_ref()
-                    .map_or(&Expression::Invalid, |e| &e.model)
+                if let Expression::Cast { from, .. } =
+                    element.upgrade().unwrap().borrow().repeated.as_ref().map_or(
+                        &Expression::Invalid,
+                        |e| {
+                            if let RepeatedElementInfo::Repeater(r) = e {
+                                &r.model
+                            } else {
+                                &Expression::Invalid
```

Comment:
Maybe `RepeatedElementInfo` should have a `ty()` function and return `Type::Component` for the non-repeater case?